src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -29,13 +41,27 @@ final class MaskingState implements ValueMaskerContext {
 
     public MaskingState(byte[] message, boolean trackJsonPath) {
         this.message = message;
+        this.messageLength = message.length;
         if (trackJsonPath) {
             currentJsonPath = new KeyMatcher.TrieNode[INITIAL_JSONPATH_STACK_CAPACITY];
         }
+        this.inputStream = null;
+        this.outputStream = null;
+    }
+
+    public MaskingState(InputStream inputStream, OutputStream outputStream, boolean trackJsonPath) {
+        this.inputStream = inputStream;
+        this.outputStream = outputStream;
+        this.message = new byte[bufferSize];
+        this.messageLength = 0;
+        if (trackJsonPath) {
+            currentJsonPath = new KeyMatcher.TrieNode[INITIAL_JSONPATH_STACK_CAPACITY];
+        }
+        readNextBuffer();
     }
 
     public boolean next() {
-        return ++currentIndex < message.length;
+        return ++currentIndex < messageLength || readNextBuffer();
```

Comment:
For readability of the code it might be nicer to make it like `|| (inputStream != && readNextBuffer()) `. Otherwise you need to navigate into readNextBuffer to see that it will always return false in that case (when you don't have an inputStream)

Reply:
I think that also would be slightly better for performance, I will do that.

Reply:
Done

Reply:
After the change, `NullAway` complains about dereferencing inputStream inside of "reloadBuffer" method. The thing is that inputStream is a class field and the analyzer does not seem to take into account the fact that this class field is null checked every time before this method is called. Also there are some other instances where "reloadBuffer" is called so that null check outside gets duplicated in 2 more places. So I figured that I keep the null check inside the method.